packages/buffered/buffered.pony

```diff
@@ -0,0 +1,35 @@
+"""
+# Buffered Package
+
+The Buffered package provides two classes, `Writer` and `Reader`, for
+writing and reading messages using common encodings. These classes are
+useful when dealing with things like network data and binary file
+formats.
+
+## Example program
+
+```pony
+use "buffered"
+
+actor Main
+  new create(env: Env) =>
+    let reader = Reader
+    let writer = Writer
+
+    writer.u32_be(42)
+    writer.f32_be(3.14)
+
+    let b = recover iso Array[U8] end
+
+    for x in writer.done().values() do
+      b.append(x)
```

Comment:
Eek! That lack of symmetry is rather irksome, isn't it?

I'd argue that we need to fix `Reader.append` then to accept a `ByteSeq`.  A quick excursion through the code suggests that it can be done without breaking any user code, with the following diff:
```diff
diff --git a/packages/buffered/reader.pony b/packages/buffered/reader.pony
index f3f37b5..e5c91bd 100644
--- a/packages/buffered/reader.pony
+++ b/packages/buffered/reader.pony
@@ -61,9 +61,9 @@ class Reader
   """
-  embed _chunks: List[(Array[U8] val, USize)] = _chunks.create()
+  embed _chunks: List[(ByteSeq, USize)] = _chunks.create()
   var _available: USize = 0
-  var _search_node: (ListNode[(Array[U8] val, USize)] | None) = None
+  var _search_node: (ListNode[(ByteSeq, USize)] | None) = None
   var _search_len: USize = 0
 
   fun size(): USize =>
@@ -79,7 +79,7 @@ class Reader
     _chunks.clear()
     _available = 0
 
-  fun ref append(data: Array[U8] val) =>
+  fun ref append(data: ByteSeq) =>
     """
     Add a chunk of data.
     """
@@ -547,13 +547,13 @@ class Reader
     end
 
     var node = if _search_len > 0 then
-      let prev = _search_node as ListNode[(Array[U8] val, USize)]
+      let prev = _search_node as ListNode[(ByteSeq, USize)]
 
       if not prev.has_next() then
         error
       end
 
-      prev.next() as ListNode[(Array[U8] val, USize)]
+      prev.next() as ListNode[(ByteSeq, USize)]
     else
       _chunks.head()
     end
@@ -574,7 +574,7 @@ class Reader
         break
       end
 
-      node = node.next() as ListNode[(Array[U8] val, USize)]
+      node = node.next() as ListNode[(ByteSeq, USize)]
     end
 
     _search_node = node
```